File: contrib/jsimplemodule/example/Sink.java
import org.omnetpp.simkernel.*;

public class Sink extends JSimpleModule {

    public Sink(long cptr) {
        super(cptr);
    }

    public void handleMessage(cMessage msg) {
        ev().println(msg.getName()+" arrived at sink");
        msg.delete();
    }

    protected void finish() {
        ev().println("finish of "+getFullPath());
    }

};

File: test/anim/multisend/sink.cc
#include <omnetpp.h>

using namespace omnetpp;

class Sink : public cSimpleModule
{
  protected:
    virtual void handleMessage(cMessage *msg) override;
};

Define_Module(Sink);

void Sink::handleMessage(cMessage *msg)
{
    delete msg;
}

